### PR TITLE
Goon-style Enzymatic Reclaimer & Cloning

### PR DESCRIPTION
**Pull request by tigercat2000**
This changes the cloning process in the following critical ways:
 - You can no longer load clonemeat directly into the cloner. Instead,
 you must use a new machine called the Enzymatic Reclaimer.
 - Biomatter is now measured out of a cap of 100.
 - Biomatter is no longer consumed all at once when cloning someone.
 Instead, it is slowly drained as long as they are being cloned. This
 means that faster cloners also use less biomatter.
 - There is a new machine called the Enzymatic Reclaimer. To use this
 machine, you must first load one of the following into it:
   - Human, alive or dead (by grab)
   - Monkey (by grab)
   - Xenomorph (by grab)
   - Some type of meat
   - Organs
 - Following that, you simply click the machine to grind up whatever you
 put inside of it. It will take anywhere from 4 to 32 seconds, depending
 on what you shoved inside, and then distribute the resultant meat to
 any cloners within a 4 tile square radius around it.

Handy tables on the numbers for the new machine and biomatter
requirements:

Reclaimer Products | Cycles  | Seconds  | Meatslurry/blood | End Biomatter
----------         | ------- | -------- | ---------------- | --------
Human              | 8 to 16 | 16 to 32 | 16 to 32         | 20 to 48
Xenomorph          | 8 to 16 | 16 to 32 | 16 to 32         | 20 to 48
Monkey             | 4 to 8  | 8 to 16  | 8 to 16          | 12 to 24
Meat/Organ         | 2 to 8  | 4 to 16  | 4 to 16          | 6 to 24

Excess product of the Reclaimer is discarded if it cannot be distributed.
The cloner uses less biomass by tier, proportional to how fast it goes.

Cloner Tier | Biomass Required Per Cloner
----------- | ---------------------------
Tier 1      | ~77
Tier 2      | ~45
Tier 3      | ~39
Tier 4      | ~29

:cl:
del: Removed adding meat directly to cloners.
add: Added a new machine from goonstation, the Enzymatic Reclaimer. Load some meat or people into it to make clones.
tweak: Cloners use way more biomass, but distribute it across the entire cycle. You can load more mid-cycle with the new reclaimer.
/:cl:

Oh, also, map edits:

### CODE REVIEW

File: code/game/machinery/cloning.dm [hunk 1]
```diff
@@ -284,32 +303,46 @@
...
+			connected_message("Clone Ejected: Low Biomatter.")
+			return
 
-		else if(occupant.cloneloss > (100 - heal_level))
+		if(meat_level <= 0)
+			failed_tick_counter++
+			if(failed_tick_counter == (MAX_FAILED_CLONE_TICKS / 2)) // halfway to ejection.
+				connected_message("Low Biomatter: Preparing to Eject Clone")
+				announce_radio_message("Low Biomatter: Preparing to Eject <b>[occupant]</b>")
+				use_power(200)
```
Comment (Crazylemon64) [inline code/game/machinery/cloning.dm]: What's with this arbitrary power consumption here?

Comment (tigercat2000) [inline code/game/machinery/cloning.dm]: it's idle power usage since the actual idle power usage won't ever be called if it returns here

**Comment by FlattestGuitar** [comment]
Does this mean you can murder someone without leaving any trace within 16 seconds?

**Comment by tigercat2000** [comment]
@FlattestGuitar actually it's supposed to make them shit out all of their gear on the floor but i forgot to port that

**Comment by FlattestGuitar** [comment]
That still leaves zero solid evidence they were murdered there and all you need is an aggressive grab. Even the chaplain's cremator won't work on almost all non-restrained targets. 

I'd make it impossible to put a live player-controlled human in there.

**Comment by tigercat2000** [comment]
@FlattestGuitar
>the chaplain's cremator won't work on almost all non-restrained targets

the fuck are you on about
this is literally no different 
they can move in any direction to get out instantly
this is actually if anything less powerful than the gibber because there's no two second delay where they can't do anything

**Comment by ktoma36** [comment]
Oh god I just had a bad thought. I play slime people a lot and there are far too many situations where doctors will try to clone me, fail, then instantly morgue me, effectively taking me out of the round. With this PR careless doctors could permanently remove a slime person from the round pretty easily. At least in the morgue there is a chance someone who knows what they are doing will notice the green light on the casket. With this though, all I would be able to do is scream in deadchat as the doc stuffs my body into the reclaimer.

Maybe make the cloning machine display more information on whats in the scanner. Make it give specific information on the person inside it instead of just the generic mental interface connection error. Its far to easy to mistake an unclonable humanoid race being unable to scan with a person who simply does not want to return to their body.

**Comment by jughu** [comment]
Inb4 ian lisa creating cubs running into a conveyor belt trough a emagged gibber and put into the reclaimer

**Comment by KasparoVy** [comment]
Don't hugbox a great feature, looks dope as-is.

**Comment by SamHPurp** [comment]
I'm struggling to find a feature of this I like.

**Comment by ktoma36** [comment]
I do like this feature as it requires people to constantly maintain the cloning machine as to dumping a lot of bio-matter into it roundstart and letting it whirl away. I just think there needs to be something baring your generic chucklefuck doctor from reclaiming a relivable person. I also feel that it will cut the coroner out of his job a bit as people are encouraged to scrap the bodies as soon as they are scanned for more bio-matter. I think to properly introduce this new mechanic there would need to be an update to the SOP for medical. Maybe even restrict cloning to only geneticist and add a geneticist job slot to help compensate for this. Geneticist are already supposed to have full control over cloning, yet I never see a geneticist manning the cloning computer.

Oh and if we are adding more features, something that I liked from goonstation is the integrated UI and SE scanners that you could optionally turn on for the cloning machine. These would automatically save a copy of your UI and SE to the database and allow the geneticist to see your genetic make up. In goon this is really nice to check if a person had some innate super powers that could easily be activated with some effort. In paradise though it would not be too useful other than to be able to turn a monkey into a carbon copy of someone for organ transplants (wonder when blood type will finally come in to effect when transplanting organs).

Overall it would seem more logical to restrict access to genetics with such a potentially destructive device implemented and would (hopefully) encourage geneticist to stick around their work place instead of running off to be telepathic annoyances.

**Comment by Anasari** [comment]
I think a safety feature that can be disabled by emag would be great, to prevent live humans from being constantly grinded up by inexperienced doctor. 

It would be like the new SR exploding.

**Comment by ktoma36** [comment]
Its not just live people, in the case I am concerned about is a doctor working too fast and ending up grinding a dead slime person through inexperience/routine . I know the goon style reclaimer doesn't leave a brain or anything (unless you are changing that).

**Comment by johannhawk** [comment]
I would like to point out that goonstation has a safety lock on their enzymatic reclaimers that are removed by emag.
Anyways, i want to know if building new reclaimers on the station to make custom built cloners is doable, and it probably is. Just checking.

**Comment by Taaketa** [comment]
What is the correct cloning proceedure for slimes anyway?

**Comment by ktoma36** [comment]
Strange reagent or transfer their slime core into another creatures head. Cloning is impossible do to the make up of a slime persons body.

**Comment by tigercat2000** [comment]
@johannhawk That was added some time after the last I played goon. I opted not to port it. It seems unnecessary to me, and we generally follow the policy of allowing dangerous machines to be, well, dangerous. I do however see the points about this deleting items and leaving no trace, so I'll add either an abiotic (must be naked and holding nothing) check for the victim, or make it forcefully drop all of their items.

**Comment by SamHPurp** [comment]
Goon added a safety feature, that you feel is unnecessary... I don't even know how to... what...

**Comment by Crazylemon64** [comment]
Encouraging dangerous workplace accidents is always a :+1: by me

**Comment by Fox-McCloud** [comment]
After having thought about this a lot over the past few days, I'm probably going to give this a 👎.

Here's my reasons:

- Can't rebuild cloning now (bombing cloning will be a viable option for traitors no matter what)
- No one else can construct cloning for personal reasons or for another department/location
- Having seen this in action a lot on Goon, it leads to way too costly of mistakes; perfectly good doctors end up grinding up bodies, permanently, who could have otherwise been cloned. Players and admins alike have a hissy fit when the Chaplain/Chef burns/gibs bodies that were potentially cloneable; this is institutionalizing it
- It's awfully easy to permanently remove someone from the round, with this, with no evidence (and tons of credibility/plausible deniability).
- People were extremely unhappy with the concept of strange reagent gibbing someone, randomly, if you mess up (and that's recoverable from). If someone messes up with this, the player is removed from the round, forever.
- There's going to be a lot of "couldn't clone them, into the grinder they go", just because the person couldn't quite jump to their corpse in time.


I think its' an interesting feature to be sure, but after giving it thought and recalling all the memories I've had of this on Goon where players were unintentionally and permanently screwed over because of an accident were...well, way too high (and this is on a server where the average round time is 40-50 minutes).

File: code/game/machinery/cloning.dm [hunk 2]
```diff
@@ -284,32 +303,46 @@
 
 	H.suiciding = 0
 	attempting = 0
+
+	if(reagents && reagents.total_volume)
```
Comment (Fox-McCloud) [inline code/game/machinery/cloning.dm]: There's no way to get reagents into the clonepod on our server.

Comment (tigercat2000) [inline code/game/machinery/cloning.dm]: The enzymatic reclaimer shoves all of it's reagents into cloning pods, not just biomatter.

**Comment by tigercat2000** [comment]
@Fox-McCloud I'm just going to close the PR because I'm really not interested in implementing all of this, but all of your points are pretty easy to address.
 - A way to make the reclaimer could be made, just another machine board.
 - Same thing.
 - It could give doctors a warning before the grind any clone-able bodies, and to address a few others, prevent NOCLONE races from being stuffed into it.
 - It's really not any easier than the crematorium or gibber. If anything, it's harder, at least with it's default location- The only point I see in this is that it is easier to use on unrestrained targets, but how often is that really going to end up being relevant? Either way, it'd be trivial to add a target.abiotic(1) check, make it the same as the gibber.
 - If someone gives a corpse to the cook, they're just as fucked. Honestly, it's not even obvious that you can stuff someone into this machine unless you already know how it works. It doesn't work like pretty much any other medbay machine, you can't just drag and drop into it.
 - Again, it's really no different than a doctor giving away bodies to the chaplain as soon as they fail cloning once- a fuck up on the doctor's part. I suppose it removes the middleman, but still.

**Comment by Aurorablade** [comment]
:(

**Pull request closed without merging**